server/src/main/java/io/crate/analyze/AnalyzedColumnDefinition.java

```diff
@@ -318,30 +322,29 @@ Settings builtAnalyzerSettings() {
         return analyzerSettings;
     }
 
-    private static void applyAndValidateStorageSettings(Map<String, Object> mapping,
-                                                        AnalyzedColumnDefinition<Object> definition) {
+    public static boolean docValuesSpecifiedAndDisabled(AnalyzedColumnDefinition<Object> definition) {
         if (definition.storageProperties == null) {
-            return;
+            return false;
         }
         Settings storageSettings = GenericPropertiesConverter.genericPropertiesToSettings(definition.storageProperties);
         for (String property : storageSettings.names()) {
             if (property.equals(COLUMN_STORE_PROPERTY)) {
                 DataType<?> dataType = definition.dataType();
                 boolean val = storageSettings.getAsBoolean(property, true);
                 if (val == false) {
-                    if (dataType.id() != DataTypes.STRING.id()) {
+                    if (dataType.id() != DataTypes.STRING.id()) { // maybe even exclude text and leave only keyword
```

Comment:
this is not really related, can be done as a follow up (I mean forbid using columnstore flag on fulltext or at least update docs to explicitly mention that doc values are not supported on fulltext indixed fields)

see also https://crate.slack.com/archives/CBPLN6WJK/p1659602904214459